L1Trigger/L1TGlobal/src/AXOL1TLCondition.cc

```diff
@@ -90,10 +90,27 @@ const bool l1t::AXOL1TLCondition::evaluateCondition(const int bxEval) const {
   int useBx = bxEval + m_gtAXOL1TLTemplate->condRelativeBx();
 
   //HLS4ML stuff
-  std::string AXOL1TLmodelversion = m_AXOL1TLmodelversion;  //config loading method
+  std::string AXOL1TLmodelversion = m_AXOL1TLmodelversion;  //loading from menu
+
+  //if model version is empty, return false. Should not happen unless menu version invalid
+  if (AXOL1TLmodelversion == "" || AXOL1TLmodelversion == "GTADModel_") {
+    edm::LogWarning("AXOL1TLCondition") << "Warning: AXOL1TL model version empty, not evaluating condition!"
+                                        << std::endl;
+    return false;
```

Comment:
Shouldn't that be caught earlier already? And/or throw an exception with a meaningful message, rather than silently returning 'false'? Under which circumstances can this happen with v1_0_1 and up?

Reply:
@quinnanm @aloeliger , I discussed this briefly with HLT experts and they prefer to have an exception here instead of returnin false as decision. This should help us catch mistakes when integrating new menus. Would it be possible to change this implementation in one of the following ways?

1.  Change GT emulator such that old UTM and new UTM works properly and gets AXO decision correctly regardless of the UTM version. i.e use v3 model as default with the old ADT score logic and get the model version if using the new UTM with AXO score logic

2. If it's not possible to do that in a short time scale, change the above logic to give an error about "incompatible menu version" and stop the job by throwing an exception.

Reply:
> @quinnanm @aloeliger , I discussed this briefly with HLT experts and they prefer to have an exception here instead of returnin false as decision. This should help us catch mistakes when integrating new menus. Would it be possible to change this implementation in one of the following ways?
> 
> 1. Change GT emulator such that old UTM and new UTM works properly and gets AXO decision correctly regardless of the UTM version. i.e use v3 model as default with the old ADT score logic and get the model version if using the new UTM with AXO score logic
> 2. If it's not possible to do that in a short time scale, change the above logic to give an error about "incompatible menu version" and stop the job by throwing an exception.

At this point I have no idea whether that's possible or not anymore. I don't know if there's a direct test for UTM version, so @quinnanm could just test for the old ADScore cut and the AXOScore cut for making AXO conditions? If the old testing cut hasn't been repurposed? I'm sort of surprised by this decision, because the original impetus of all our discussions of adding fail-safes like the above were to prevent possible HLT failures in online use.

Reply:
Hmm, the desired behaviour is that a) the code should not crash, b) if there is something, an exception should be thrown, with meaningful error message, especially in case of a job setup or config error. I assume this can not change from one event to the next within a job (ie, change only with a change of L1T menu).

Reply:
ok, I will make a new commit now that throws an exception instead.

Reply:
as of more recent commits, this is now an exception as requested